### PR TITLE
Describe async closures

### PR DESCRIPTION
**Pull request by compiler-errors**
r? @ehuss

Stabilization PR has all the info here: https://github.com/rust-lang/rust/pull/132706

More technical details on async closures here: https://rustc-dev-guide.rust-lang.org/coroutine-closures.html

I'm not certain how detailed I need to go here. I'm also not certain if I should edit the `src/types/closure.md` chapter to discuss when async closures implement `AsyncFn*` and `Fn*`.

### CODE REVIEW

**Review by traviscross — APPROVED** [review]

File: src/types/closure.md
Comment (compiler-errors) [inline src/types/closure.md:499]:
```suggestion
environment. Non-async, non-capturing closures can be coerced to function pointers (e.g., `fn()`)
```

File: src/types/closure.md
Comment (compiler-errors) [inline src/types/closure.md:524]:
```suggestion
If the async closure is lending to its `Future`, then [`FnMut`] and [`Fn`] are *not* implemented. `FnOnce` is always implemented.
```

**Review by compiler-errors — COMMENTED** [review]
LGTM

**Review by ehuss — APPROVED** [review]
Thanks! Will merge after stabilization is merged.

**Comment by compiler-errors** [comment]
Stabilization PR has landed, so this is ready to go.

**Pull request merged**
